### PR TITLE
enhance(std.fmt.parseInt): may returns null for ?T

### PR DESCRIPTION
**Pull request by pseudocc**
Empty string is invalid for non-optional types, while for the optional types, it is valid and the function will return null.

### CODE REVIEW

File: lib/std/fmt.zig [hunk 1]
```diff
@@ -1733,12 +1733,23 @@ pub fn Formatter(comptime format_fn: anytype) type {
 /// Ignores '_' character in `buf`.
 /// See also `parseUnsigned`.
 pub fn parseInt(comptime T: type, buf: []const u8, base: u8) ParseIntError!T {
-    if (buf.len == 0) return error.InvalidCharacter;
+    if (buf.len == 0) {
+        if (comptime isOptional(T)) return null;
```
Comment (travisstaloch) [inline lib/std/fmt.zig:1737]: if think this could be simplified to 
```zig
if (@typeInfo(T) == .Optional) return null;
```

Comment (pseudocc) [reply lib/std/fmt.zig:1737]: That's cool, I am new to zig, thanks for the advice.

**Comment by andrewrk** [comment]
I like status quo better. Just put `catch null` at your call site to get this behavior.

**Comment by pseudocc** [comment]
> I like status quo better. Just put `catch null` at your call site to get this behavior.

Hi @andrewrk , my thought is that, currently `parseInt(i32, string)` and `parseInt(?i32, string)`
behave the same way, they all never return null, and I think `catch null` is a little bit overkill
in some of the way, how about we add another error type, say `EmptyString`, for this.

If the user want to keep the status quo, they can use `var num: ?i32 = parseInt(i32, string)`,
just my two cents, thanks.

**Pull request closed without merging**